### PR TITLE
feat(ui): Storybook restructuring

### PR DESCRIPTION
**Pull request by vuluongj20**
Two updates to Storybook: new categories ("Core", "Assets", "Components", "Views",…), and new docs-ui folder structure.

**1. New categories**
The stories are now organized into more intuitive and appropriate categories, including: 
 - "Core"
 - "Assets"
 - "Components"
 - "Views"
 - "Utilities"
 - "Features"
 - "Deprecated"
Read `docs-ui/stories/core/overview.stories.mdx` to learn more about what each category contains.

**2. New docs-ui folder structure**
docs-ui now has a new folder structure:
 - docs-ui/stories: new home for .stories.* files. All stories are further organized into sub-folders according to their respective category (stories/core, stories/assets,…)
 - docs-ui/images: for accompanying images that will be used in future .mdx stories
 - docs-ui/components: for reusable, Storybook-specific components that will be used in future .mdx stories. For example: a DO/DON'T box that will reappear in many stories.
 - docs-ui/less: same as before - for less files

### CODE REVIEW

File: .storybook/manager.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import {create} from '@storybook/theming';
 
 const theme = create({
   base: 'light',
-  brandTitle: 'Sentry Styleguide',
+  brandTitle: 'Sentry UI System',
```
Comment (vuluongj20) [inline .storybook/manager.ts:6]: "UI system" seems appropriate here because we're referring to a group of UI assets/components that are related to one another though a common design language (hence "system"). This may change if we find a more appropriate name.

File: babel.config.ts [hunk 2]
```diff
@@ -20,6 +20,17 @@ const config: TransformOptions = {
     ],
     '@babel/preset-typescript',
   ],
+  overrides: [{
```
Comment (vuluongj20) [inline babel.config.ts:23]: This override is necessary because I ran into a runtime error from babel:
`importSource cannot be set when runtime is classic`

We are using React 17's new JSX transform with automatic runtime (see line 10 of this same file). Further, we have set importSource to be `@emotion/react` (see line 11). The problem is that the importSource option is only available to automatic runtime, and for some reason our Storybook compiling falls back to classic runtime (perhaps due to lack of support for React 17 from the version of Storybook that we're using). So, the importSource command cannot be fulfilled, leading to the error.

The solution: set an overriding Babel config object only applies to Storybook, telling Babel to use automatic runtime but with no specific, developer-defined importSource. This resolves the issue above.

File: docs-ui/stories/components/form-fields.stories.js [hunk 3]
```diff
@@ -374,4 +373,4 @@ export const __TextCopyInput = () => (
   <TextCopyInput onCopy={action('Copied!')}>Value to be copied </TextCopyInput>
 );
 
-__TextCopyInput.storyName = 'TextCopyInput';
+__TextCopyInput.storyName = 'Text Copy';
```
Comment (billyvg) [inline docs-ui/stories/components/form-fields.stories.js:376]: Should story name match the component name exactly?

File: package.json [hunk 4]
```diff
@@ -108,6 +109,7 @@
     "prism-sentry": "^1.0.2",
     "process": "^0.11.10",
     "prop-types": "^15.6.0",
+    "puppeteer": "^10.1.0",
```
Comment (billyvg) [inline package.json:112]: Why is this needed?

File: docs-ui/stories/components/form-fields.stories.js (hunk 3 above)
Comment (vuluongj20) [reply docs-ui/stories/components/form-fields.stories.js:376]: You can still look up TextCopy and it'll show you this story. However, if the story name is too different from the component name (which is currently a bit rare), then it won't show up.

I'll likely revise the Storybook sidebar. Among other things, it'll allow people to search for stories using metadata such as name, component name, and even related keywords (which will be custom-defined in the story files). Storybook allows for some overwriting of their default UI and behavior. I just have to find an efficient way to do it for the sidebar.

File: package.json (hunk 4 above)
Comment (vuluongj20) [reply package.json]: Oh shoot this is from when I was trying to set up Storybook and got a compilation error. Just removed it :)

**Review by billyvg — APPROVED** [review]
